KubeArmor/monitor/systemMonitor.go

```diff
@@ -245,139 +245,133 @@ func (mon *SystemMonitor) InitBPF() error {
 		}
 	}
 
-	bpfPath := homeDir + "/BPF/system_monitor.c"
-	if _, err := os.Stat(filepath.Clean(bpfPath)); err != nil {
+	bpfObjPath := homeDir + "/BPF/objs/"
+	if _, err := os.Stat(filepath.Clean(bpfObjPath)); err != nil {
 		// go test
 
-		bpfPath = os.Getenv("PWD") + "/../BPF/system_monitor.c"
-		if _, err := os.Stat(filepath.Clean(bpfPath)); err != nil {
+		bpfObjPath = os.Getenv("PWD") + "/../BPF/objs/"
+		if _, err := os.Stat(filepath.Clean(bpfObjPath)); err != nil {
 			return err
 		}
 	}
 
-	content, err := ioutil.ReadFile(filepath.Clean(bpfPath))
-	if err != nil {
-		return err
-	}
-	bpfSource := string(content)
-
 	mon.Logger.Print("Initializing an eBPF program")
 
 	if mon.EnableKubeArmorPolicy && !mon.EnableKubeArmorHostPolicy { // container only
-		mon.BpfModule = bcc.NewModule(bpfSource, []string{"-O2"})
-		if mon.BpfModule == nil {
-			return errors.New("bpf module is nil")
+		mon.BpfModule, err = lbpf.OpenObjectFromFile(bpfObjPath + "system_monitor.bpf.o")
```

Comment:
Does this function call assume that the object file is already there(ie the bpf code is already compiled)?

Reply:
Sorry didn't see this before. Yes it does assume that